odoo/addons/base/models/res_country.py

```diff
@@ -106,7 +111,10 @@ def create(self, vals_list):
     def write(self, vals):
         if vals.get('code'):
             vals['code'] = vals['code'].upper()
-        return super(Country, self).write(vals)
+        res = super(Country, self).write(vals)
+        self.flush()
+        self.clear_caches()
```

Comment:
unless I'm missing something, you don't need to clear the cache *unless* `code` or `phone_code` was changed, right?

You never know, someone could add a dumb computed field on `res.country` that frequently causes `write()`, and you want to avoid cache invalidations as much as possible because they're expensive ;)

BTW you could consider that your cache management is incomplete, as you do not handle the cases of creation/deletion of countries. But for the majority of the cases this should be sufficient, so I'd tend to voluntarily ignore the rest for the sake of less code. (maybe with a comment in the `write` case to say that create/unlink cases are ignored on purpose).